src/server/cmd/pachd/main.go

```diff
@@ -371,234 +369,219 @@ func doFullMode(config interface{}) (retErr error) {
 		return fmt.Errorf("ListenAndServe: %v", err)
 	})
 	eg.Go(func() error {
-		_, grpcEg := grpcutil.Serve(
```

Comment:
So this is getting rid of the `Serve` function, it's a little hard for me to tell what's really replacing it and what it's called, I don't see a `Serve` method in the replacement code. Could you describe what the equivalent is in the new code? And is it possible to still have a method called `Serve`? That to me seems like the most understandable api for starting up the server.

Reply:
Alright I think I understand this better after reading a bit more, you basically call `NewServer` to get back a server and then call `Wait` on it when you've registered. Does `Wait` do anything other than just waiting for the server to exit?

Reply:
Here's an example:

```go
// create a server
// second argument specifies whether TLS is used
server, err := grpcutil.NewServer(context.Background(), false)`
// listen on TCP port 30650
_, err = server.ListenTCP("", 30650)
// wait until exit
server.Wait()
```

I renamed it to `ListenTCP` because originally I had added `ListenUDS`. Even though that's removed, I figured it's useful in case future protocols are introduced. I'm happy to remove it otherwise though.

Reply:
The old equivalent of the above would've been:

```go
_, eg := grpcutil.Serve(
    context.Background(),
    grpcutil.ServerOptions{
        MaxMsgSize: grpcutil.MaxMsgSize,
        Port:       30650,
        RegisterFunc: func(s *grpc.Server) error {
            ...
            return nil
        },
    },
)
eg.Wait()
```